### DISCUSSION TITLE
[Task]: Inference on a test set

### DISCUSSION THREAD

**Discussion by dangmanhtruong1995** [open-edge-platform/anomalib#861] [Q&A]
### What is the motivation for this task?

The example on the github page shows only inference for an image. 

### Describe the solution you'd like

If anyone know how to do inference on a test set that would be great.

**Comment by samet-akcay** [comment]
Hi @dangmanhtruong1995, `--input` and `--output` flags checks whether it is an input to an image or folder containing images. Assuming that you train a `padim` model on `MVTec bottle` category, you could run `torch_inference.py` to perform an inference on a number of images from a directory.

```bash
python tools/inference/torch_inference.py \
    --config anomalib/models/padim/config.yaml \
    --weights results/padim/mvtec/bottle/run/weights/model.ckpt \
    --input datasets/MVTec/bottle/test/broken_large \    # Path to broken_large images in the test set.
    --output output/broken_large    # An output directory to save the predicted images.
```

Let me know if this is what you are looking for.

**Comment by samet-akcay** [comment]
I'm converting this to a Q&A in discussions. Let's continue there.

**Comment by dangmanhtruong1995** [comment]
Hi @samet-akcay , so I have a test folder containing images in "abnormal" and "normal" subfolders, I want to do inference on all of those images and then output F1 score, AUC, .... as when we trained before.

**Comment by samet-akcay** [comment]
Hm, sounds like [`tools/test.py`](https://github.com/openvinotoolkit/anomalib/blob/main/tools/test.py) might suit better in this case.

**Reply by dangmanhtruong1995** [reply]
> Hm, sounds like [`tools/test.py`](https://github.com/openvinotoolkit/anomalib/blob/main/tools/test.py) might suit better in this case.

So how do I run using tools/test.py ? Because on the readme it does not talk about this.

**Comment by samet-akcay** [comment]
https://github.com/openvinotoolkit/anomalib/blob/main/tools/test.py#L23-L25

You will need to provide `--model` or `--config` and `--weight_file` as arguments, such as the following:

```bash
python tools/test.py \
    --config </path/to/your/config/file>
    --weight_file </path/to/your/weight/file>
```

if you use this script, make sure that your the data that you want to test follows the same structure as specified in your config file (mvtec, btech, folder etc).

**Reply by dangmanhtruong1995** [reply]
Hi, I have managed to run it as you said, thanks a lot.
